api/utils/rights.js

```diff
@@ -496,50 +495,74 @@ function wrapCallback(params, callback, callbackParam, func) {
 * @param {function} callback - callback method
 **/
 function dbLoadEventsData(params, apps, callback) {
+    const appIds = [];
+    const appNamesById = {};
+
+    apps.forEach((app) => {
+        appIds.push(common.db.ObjectID(app._id + ''));
+        appNamesById[app._id + ''] = app.name;
+    });
 
     /**
     * Get events collections with replaced app names
     * A helper function for db access check
-    * @param {object} app - application object
+    * @param {object} appColl - application ids and names
     * @param {function} cb - callback method
     **/
-    function getEvents(app, cb) {
+    function getEvents(appColl, cb) {
         var result = {};
-        common.db.collection('events').findOne({'_id': common.db.ObjectID(app._id + "")}, function(err, events) {
-            if (!err && events && events.list) {
-                for (let i = 0; i < events.list.length; i++) {
-                    result[crypto.createHash('sha1').update(events.list[i] + app._id + "").digest('hex')] = "(" + app.name + ": " + events.list[i] + ")";
+        common.db.collection('events').find({'_id': { $in: appColl.appIds }}).toArray(function(err, events) {
+            appColl.appIds.forEach((appId) => {
```

Comment:
Lets say you are getting events for 5 apps.
Then in events you have list of 5 documents - one for each app. Lets assume each app has 5 events.

Then here as you do for each APP - it will process whole array for each app  - creating  records for all events(not just ones for this app)

So you would get 5*(5*5) = 125 , not 25 records as you should.